Utility/WikiObjectPropertiesListGenerator.FCMacro

```diff
@@ -0,0 +1,401 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# (c) 2021 <Evgeniy> LGPL
+# Modified by <hasecilu> on 2024-11-14
+
+"""
+Description:
+This macro generates lists of properties of the selected object in different forms
+in Wiki format. The list is printed to the Report view of FreeCAD. This macro can
+be used to simplify working with this Wiki with some restrictions. The property
+lists published in Wiki usually contain only unique parameters. Parameters inherited
+from parent objects are not published.
+
+This script automatically removes inherited parameters from the list. But due to the
+software limitations of FreeCAD, it cannot delete absolutely all inherited parameters,
+so the lists generated by this script should be carefully checked before publishing.
+
+Usage:
+- Preparing:
+    - Copy code of Macro from this page and place it in a new *.FCMacro file.
+    - Save FCMacro file, and place it in the FreeCAD Macros folder.
+    - Disable Include a timecode for each entry checkbox in Preferences -> General -> Output window (Tab).
+
+- First using:
+    - Select (or create and select) only one object in tree view the type of you need.
+    - Select Macro -> Macros... item in main menu.
+    - In opened window find *.FCMacro file what you save letter and press Execute Button.
+    - In the dialog that opens, select the type of report that you want to get by pressing button:
+        - Only Main - prints only unique properties of base (parent) classes.
+        - All Properties - prints absolutely all properties of selected object.
+        - Inherits+Unique Properties - prints only unique properties with included inherits report.
+        - Only Unique Properties - prints only unique properties.
+
+- How to use again:
+    - To quickly run this macro again, you can use keyboard shortcut `Shift+Ctrl+1` it will launch the last run macro.
+"""
+
+__Name__ = "Wiki Object Properties List Generator"
+__Comment__ = "Generates lists of object properties for use in the FreeCAD Wiki documentation."
+__Author__ = "Evgeniy"
+__Date__ = "2024-11-15"
+__Version__ = "0.16"
+__License__ = "LGPL-3.0-or-later"  # 6 matches for LGPL on https://spdx.org/licenses/
+__Web__ = ""
+__Wiki__ = "https://wiki.freecad.org/Macro_Wiki_Object_Properties_List_Generator"
+__Icon__ = "./WikiObjectPropertiesListGenerator.svg"
+__Xpm__ = ""
+__Help__ = "Select a scripted objects on the tree and execute the macro."
+__Status__ = "Beta"
+__Requires__ = ""
+__Communication__ = "https://forum.freecad.org/viewtopic.php?f=21&t=61998"
+__Files__ = "./WikiObjectPropertiesListGenerator.svg"
+
+# NOTE: Changelog
+# ---------
+# - 0.16 - 2024-11-15
+#   - Refactor code: add custom dataclass, update functions' signatures, make some fixes.
+#   - Include property's value on the list.
+#   - Create custom dialog to add checkbox to let user disable/enable property status.
+# - 0.15 - 2021-09
+#   - Initial release.
+
+import re
+
+import FreeCAD
```

Comment:
Please use `import FreeCAD as app` and `import FreeCADGui as gui` to be consistent with most other macros.